src/queue/song.py

```diff
@@ -0,0 +1,42 @@
+class Song:
+
+    def __init__(self, code, votes, ip):
+        self.code = code
+        self.votes = votes
+        self.upvoted = [ip]
```

Comment:
Is this assuming when someone adds a song that they are also upvoting it?

Reply:
Yes, that's the implementation I suggest for the time being.  Similar to reddit for upvoting.